src/librustc_mir/interpret/terminator.rs

```diff
@@ -249,7 +249,9 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
 
         match instance.def {
             ty::InstanceDef::Intrinsic(..) => {
-                if caller_abi != Abi::RustIntrinsic {
+                if let Abi::RustIntrinsic | Abi::PlatformIntrinsic = caller_abi {
```

Comment:
I think we can just remove this entire check. Since you can't create a function pointer to intrinsics, you can't cause an ABI mismatch without rustc being severly broken

Reply:
cc @eddyb I think the plan was to, at some point, unify all of these ABIs somehow, but I don't know if somebody is already working on that.

Reply:
This is dead code our redundant checking depending on the way you look at it. I'd be fine with just removing the check. I can't come up with a way to trigger it, so returning an error seems quite wrong anyway.

Reply:
>  I can't come up with a way to trigger it, so returning an error seems quite wrong anyway.

Well this PR triggers it if one removes the `| Abi::PlatformIntrinsic` part. I suspect that since only `"rust-intrinsic"` is accepted, one can also trigger it using `"rust-call"`.

Reply:
You can't call an intrinsic (we are in the InstanceDef::Intrinsic arm) with "rust-call" ABI. I'm very certain this is impossible due to

https://github.com/rust-lang/rust/blob/acdf83f2288e1b80259dafeca4a0cee9a42973c3/src/librustc/ty/instance.rs#L150-L158

actually doing the exact same check of the one you're doing now.